snapcraft/internal/lifecycle/_containers.py

```diff
@@ -45,16 +45,33 @@ def containerbuild(step, project_options, output=None, args=[]):
                 metadata=config.get_metadata()).execute(step, args)
 
 
-def cleanbuild(project_options, remote=''):
-    if remote and not lxd._remote_is_valid(remote):
-        raise errors.InvalidContainerRemoteError(remote)
+def cleanbuild(*, project, echoer, build_environment, remote='') -> None:
+    config = project_loader.load_config(project)
+    tar_filename = _create_tar_file(project.info.name)
 
-    config = project_loader.load_config(project_options)
-    tar_filename = '{}_{}_source.tar.bz2'.format(
-        config.data['name'], config.data['version'])
+    if build_environment.is_lxd:
+        _deprecated_cleanbuild(project, remote, config, tar_filename)
+        return
 
+    build_provider_class = build_providers.get_provider_for('multipass')
+    with build_provider_class(project=project, echoer=echoer) as instance:
+        instance.provision_project(tar_filename)
+        instance.build_project()
+        instance.retrieve_snap()
```

Comment:
Given the factory nature of this, the interface used here should probably be implemented by the `BaseProvider`.